core/src/main/java/com/nextdoor/bender/ipc/IpcSenderService.java

```diff
@@ -165,23 +174,25 @@ synchronized public void flush() throws InterruptedException, TransportException
       Thread.sleep(5);
     }
 
-    /*
-     * Some factories keep state on the transports they create. Perform any cleanup that is
-     * required.
-     */
-    transportFactory.close();
-
     /*
      * Collect runtime of each thread
      */
     this.getRuntimeStat().join();
 
     /*
-     * Fail if there are any errors
+     * Fail if there are any errors so we don't close the factory and end up with partial success.
      */
     if (this.hasUnrecoverableException.getAndSet(false)) {
-      throw new TransportException("Not all transports succeeded");
+      // TODO: confirm for the s3 case that clients will need a lifecycle policy
+      //  that aborts incomplete uploads within a timeframe.
+      throw new TransportException("Not all transports succeeded when sending remaining buffers in the IpcSenderService during a flush operation.");
```

Comment:
Same thing as below, need to verify that `transportFactory.close()` does get called by something.

Reply:
@stlava shouldn't reach the IPC sender's flush() method since we abort the handler immediately as soon as we discover that a thread has failed and has set unrecoverableException to true.

Reply:
Yes but `transportFactory.close()` should be called otherwise we could have dangling state. The `close()` is supposed to clean that up.

Reply:
Verified now that we call it.